lib/vocabularies/discriminator/index.ts

```diff
@@ -69,18 +69,46 @@ const def: CodeKeywordDefinition = {
           sch = resolveRef.call(it.self, it.schemaEnv.root, it.baseId, sch?.$ref)
           if (sch instanceof SchemaEnv) sch = sch.schema
         }
-        const propSch = sch?.properties?.[tagName]
-        if (typeof propSch != "object") {
+        let propSch = sch?.properties?.[tagName]
+        let hasSubSchRequired = false
+        if (!propSch && sch?.allOf) {
+          const {hasRequired, propertyObject} = mapDiscriminatorFromAllOf(propSch, sch)
```

Comment:
Re: the GitHubActions warnings, maybe we can rename this key or avoid the destructuring and just use a constant for the returned object and reference each key in the following lines?

E.g.

```js
const discriminator = mapDiscriminatorFromAllOf(propSch, sch)
hasSubSchRequired = discriminator.hasRequired
propSch = discriminator.propertyObject
```